src/seedu/addressbook/data/person/Module.java

```diff
@@ -0,0 +1,47 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+    public class Module {
+
+        public static final String EXAMPLE = "valid Module Name";
+        public static final String MESSAGE_MODULE_CONSTRAINTS =
+                "MODULE NAME SHOULD START AS 'MODULEC PREFIX' + 'MODULE CODE'";
+        public static final String MODULE_VALIDATION_REGEX = "[\\w\\.]+@[\\w\\.]+";
+        private String date;
+        private String period;
+        private String length;
+        public final String value;
+        private boolean isPrivate;
+
+        public Module(String module,boolean isPrivate) throws IllegalValueException {
+
+            this.isPrivate = isPrivate;
+            module = module.trim();
+            if (!isValidModule(module)) {
+                throw new IllegalValueException(MESSAGE_MODULE_CONSTRAINTS);
+            }
+            this.value = module;
+
+        }
+
```

Comment:
Missing header comment. All non-trivial methods should have java doc format header comments.